### PR TITLE
Fix `Agent` unintentionally modifying `stream` for all Prompt Drivers

### LINKED ISSUE #1440: SummaryConversationMemory displays summary when using Streaming
- [x] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Describe the bug**
When using SummaryConversationMemory and Streaming, the output will contain the summary of the conversation, even though it doesn't end up in the conversation memory runs.

This gets displayed to the user as something like:

"The human introduced themselves and the assistant responded".

**To Reproduce**

It happens both with Chat and with the Stream utility. The extra information isn't showing up in the logs though.

Here's how to reproduce it with streaming:

```python
from griptape.memory.structure import SummaryConversationMemory
from griptape.structures import Agent
from griptape.utils import Stream

conversation_memory = SummaryConversationMemory(max_runs=2, autoprune=True)
agent = Agent(conversation_memory=conversation_memory, stream=True)

for artifact in Stream(agent).run("hi"):
    print(artifact.value, end="", flush=True)

for artifact in Stream(agent).run("My name is jason, what's yours?"):
    print(artifact.value, end="", flush=True)
```

You'll get output like:

```bash
[12/13/24 21:53:04] INFO     PromptTask f8faec278598460dac5cd76c72190bc0
                             Input: hi
Hello! How can I assist you today?[12/13/24 21:53:05] INFO     PromptTask f8faec278598460dac5cd76c72190bc0              

                             Output: Hello! How can I assist you today?

                    INFO     PromptTask f8faec278598460dac5cd76c72190bc0
                             Input: My name is jason, what's yours?
Hi Jason! I don't have a personal name, but you can call me Assistant. How can I help you today?[12/13/24 21:53:06] INFO
     PromptTask f8faec278598460dac5cd76c72190bc0
                             Output: Hi Jason! I don't have a personal name, but you can call me Assistant. How can I   
                             help you today?

The human greeted the assistant with "hi," and the assistant responded with a friendly "Hello!" and asked how they could assist the human.
```

The organization is a bit messy, but you can see the Output doesn't have the extra message, but it comes through when streaming. If I set logging to ERROR only, you get:

```bash
Hello! How can I assist you today?
Nice to meet you, Jason! I don't have a personal name, but you can call me Assistant. How can I help you today?
The human greeted the assistant with "hi," and the assistant responded with a friendly "Hello!" and asked how they could assist the human.
```

**Expected behavior**
The summary shouldn't be output back to the user

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Fixed
- `Agent` unintentionally modifying `stream` for all Prompt Drivers.
## Issue ticket number and link
Closes #1440

### CODE REVIEW

File: griptape/structures/agent.py [hunk 1]
```diff
@@ -38,18 +39,20 @@ def validate_fail_fast(self, _: Attribute, fail_fast: bool) -> None:  # noqa: FB
     def __attrs_post_init__(self) -> None:
         super().__attrs_post_init__()
 
-        self.prompt_driver.stream = self.stream
+        # Make a copy to avoid unintentionally modifying the global default Prompt Driver's stream setting
+        prompt_driver = copy(self.prompt_driver)
+        prompt_driver.stream = self.stream
```
Comment (vachillo) [inline griptape/structures/agent.py:44]: we aren't setting `self.prompt_driver` with the copy?

Comment (collindutter) [reply griptape/structures/agent.py:44]: `self.prompt_driver` only exists as a convenient way to set the prompt drivers on the Tasks. It doesn't get used anywhere else besides here so I don't think it makes sense to overwrite it with a new value.

Comment (vachillo) [reply griptape/structures/agent.py:44]: but i could still modify the global default by doing `agent.prompt_driver.stream = True`? and then the behavior of my agent wouldnt reflect that setting change

Comment (vachillo) [reply griptape/structures/agent.py:44]: i liked this: `Factory(lambda: copy(Defaults.drivers_config.prompt_driver))`

Comment (collindutter) [reply griptape/structures/agent.py:44]: omg: https://www.attrs.org/en/stable/api.html#attrs.evolve

**Review by vachillo — APPROVED** [review]

**Pull request merged**